deny.toml

```diff
@@ -7,49 +8,44 @@ targets = [
 ]
 
 [advisories]
-vulnerability = "warn"
-unmaintained = "allow"
-unsound = "warn"
-yanked = "warn"
-notice = "warn"
+yanked = "deny"
 
 [bans]
 multiple-versions = "allow"
 
-[[bans.deny]]
-name = "time"
-version = "0.1"
-
 [licenses]
-unlicensed = "deny"
-copyleft = "deny"
-allow-osi-fsf-free = "neither"
-default = "deny"
+unused-allowed-license = "allow"
 confidence-threshold = 1.0
 allow = [
     "Apache-2.0",
+    "BSD-2-Clause",
     "BSD-3-Clause",
     "CC0-1.0",
     "ISC",
     "LicenseRef-ring",
     "LicenseRef-webpki",
     "MIT",
+    "MPL-2.0",
+    "Unicode-3.0",
     "Unicode-DFS-2016",
     "Zlib",
     "Unlicense",
-    "OpenSSL",
```

Comment:
I would prefer merging it as-is (so current state from op-templating) and raise an issue with our license-expert @lfrancke, so that we can allow this in op-templating and here at the same time

Reply:
I think operator-templating should be the source of truth for the deny.toml and where license discussions should happen

Reply:
Created https://github.com/stackabletech/operator-templating/pull/464 for the inclusion. If we merge it in time, we can directly included that in this PR

Reply:
I'll take a quick look

Reply:
Probably a dumb question, but why are there so many changes to the deny.toml in this PR?

Is this just to get the file in line with https://github.com/stackabletech/operator-templating/pull/430 in operator-templating?

Reply:
In principle the OpenSSL license seems to be non-critical, but it does expect us to credit the openssl project in advertising ..

At the very least we need to add it to the licenses folder when building the docker images for our operators (do we even package licenses there yet?)..

https://www.tldrlegal.com/license/openssl-license-openssl
https://opensource.stackexchange.com/questions/11558/what-needs-to-be-done-to-comply-with-openssl-double-license

Reply:
So, looks like the license was greenlit based on this comment: https://github.com/stackabletech/operator-rs/pull/753#issuecomment-2015244865 back in March.

The check now fails because @sbernauer changed the deny.toml to look like what we have in templating and the OpenSSL license wasn't added there back then, so now it was removed here.

I'd suggest adding it and also adding it in templating, just as @sbernauer proposed.

Reply:
@Techassi any opinions on the use of aws-lc or ring?

Reply:
Spoke with @Techassi, he's fine with either ring or aws-lc-rs

Reply:
Ok,somewhat relevant to https://github.com/stackabletech/operator-rs/pull/891

Reply:
Regarding ring, I'm keen for it, but I think since it is crypto, we maybe need some extra eyes across it to judge its fitness.

Reply:
The aws_lc_rs provider uses ring. This discussion is only about the configuration of ring (which ciphers to include).

Reply:
Actually the motivation of using ring over aws-lc-rs might be the OpenSSL license (which aws-lc-rs has).
However, looking at https://github.com/briansmith/ring/blob/main/LICENSE, ring also has parts below OpenSSL license, so that argument seems to be misleading. Looks like we need the OpenSSL license in any case, and we are just "lucky", that the ring crate does [not have machine readable license](https://github.com/briansmith/ring/blob/befdc87ac7cbca615ab5d68724f4355434d3a620/Cargo.toml#L8) compared to [aws-lc-sys](https://github.com/aws/aws-lc-rs/blob/cdb1651d7398148688fa4c0106135b4a82ec1e49/aws-lc-sys/Cargo.toml#L9).

So to sum up: I think ring isn't any better than aws-lc-rs from a license perspective. I would prefer to focus this PR on derivative and get it merged and do the ring vs aws-lc-rs discussion in https://github.com/stackabletech/decisions/issues/35